python/pyarrow/compat.pxi

```diff
@@ -31,12 +29,8 @@ def encode_file_path(path):
     return encoded_path
 
 
-if sys.version_info >= (3, 7):
-    # Starting with Python 3.7, dicts are guaranteed to be insertion-ordered.
-    ordered_dict = dict
-else:
-    import collections
-    ordered_dict = collections.OrderedDict
+# Starting with Python 3.7, dicts are guaranteed to be insertion-ordered.
+ordered_dict = dict
```

Comment:
Not necessarily here, but I suppose we can also remove `ordered_dict` alltogether, and use `dict(..)` instead in the few places it is being used?

Reply:
`ordered_dict` has a richer API than `dict`, so theoretically we cannot return `dict` instead without breaking compatibility. Not sure that matters in practice, though.

Reply:
> `ordered_dict` has a richer API than `dict`, so theoretically we cannot return `dict` instead without breaking compatibility. Not sure that matters in practice, though.

Ah, yes indeed. It's currently being used in `Table/RecordBatch.to_pydict` and `KeyValueMetadata.to_dict`. 
Personally, I would maybe just break this at some point